java8/src/main/java/io/papermc/paperclip/Paperclip.java

```diff
@@ -224,6 +232,43 @@ private static Method getMainMethod(final Path paperJar, final String mainClass)
         }
     }
 
+    private static void mavenInstall(Path paperJar) {
+        try {
+            if (new ProcessBuilder("mvn", "-version").start().waitFor() != 0) {
+                // Error! It's either not on the path, or something went terribly wrong.
+                System.err.println("Maven must be installed and on your PATH.");
+                System.exit(1);
+            }
+        } catch (IOException | InterruptedException ex) {
+            System.err.println("Maven must be installed and on your PATH.");
+            ex.printStackTrace();
+            System.exit(1);
+        }
+
+        try {
+            Path pomPath = Paths.get("paper.xml");
+
+            try(InputStream pom = Optional.ofNullable(Paperclip.class.getResourceAsStream("/META-INF/maven/io.papermc.paper/paper/pom.xml"))
+                .orElseGet(() -> Paperclip.class.getResourceAsStream("/META-INF/maven/com.destroystokyo.paper/paper/pom.xml"))) {
```

Comment:
I would prefer a normal `if` before the `try` rather than a super long `Optional` call that has a wrap.